src/plugins/discover/public/application/angular/doc_table/components/table_header/table_header_column.tsx

```diff
@@ -174,7 +174,9 @@ export function TableHeaderColumn({
                 className={button.className}
                 data-test-subj={button.testSubject}
                 onClick={button.onClick}
-              />
+              >
+                <EuiIcon {...button.iconProps} size="s" />
+              </button>
```

Comment:
Since `<button>` is not a EUI element, it leads to following behaviour. For instance, when 
moving columns, focus is not leaving at the target buttton like it was. The reason is that for non-eui elements in a focus state `outline: none` will be set. 

https://github.com/elastic/kibana/blob/373d6e2a2e132650528ae69fe585cb7c20b88a03/src/core/public/styles/_base.scss#L15

See the difference in the video. 

https://user-images.githubusercontent.com/39378793/126037549-0bb5305d-510e-4fcf-8111-a86137f345d0.mp4

In case if it's really matters, one option I see: use `EuiButtonIcon` and reset height and width to 12px instead of 24 which is defined for `xs` variant.
What do you think about it? @andreadelrio @timroes @majagrubic

Reply:
Would it just make sense to use an `EuiIconButton` instead here then, same as we switched for the pagination buttons above, with some custom styling (to make it that small)? (Maybe I am missing some background here, why this won't work or have already been tried.)

Reply:
This is exactly what I suggest to do, but in this case we will have rewrite Eui styles

Reply:
Done